libraries/aws.rb

```diff
@@ -93,7 +93,10 @@ def specific_machine_options(component, _count = nil)
         return [] unless @node['delivery-cluster'][component]
         options = []
         options << { bootstrap_options: { instance_type: @node['delivery-cluster'][component]['flavor'] } } if @node['delivery-cluster'][component]['flavor']
+        options << { bootstrap_options: { security_group_ids: @node['delivery-cluster'][component]['security_group_ids'] } } if @node['delivery-cluster'][component]['security_group_ids']
+        options << { image_id: @node['delivery-cluster'][component]['image_id'] } if @node['delivery-cluster'][component]['image_id']
         # Specify more specific machine_options to add
+        options
```

Comment:
Because of the `if` at the end of line 97, it's possible that nothing will be added to the `options` list. In that case, that entire expression returns `nil`, and thus the method.

Here's a `pry` session to demo:
``` ruby
[1] pry(main)> x = []
=> []
[2] pry(main)> y = (x << "foo")
=> ["foo"]
[3] pry(main)> y
=> ["foo"]
[4] pry(main)> z = (y << "foo" if false)
=> nil
[5] pry(main)> z
=> nil
```
So yes, we do need `options` here :smile: